partiql-lang/src/main/kotlin/org/partiql/lang/graph/ExternalGraphReader.kt

```diff
@@ -1,16 +1,26 @@
 package org.partiql.lang.graph
 
+import com.amazon.ion.IonList
+import com.amazon.ion.IonSequence
+import com.amazon.ion.IonSexp
+import com.amazon.ion.IonString
+import com.amazon.ion.IonStruct
+import com.amazon.ion.IonSymbol
 import com.amazon.ion.IonSystem
 import com.amazon.ion.IonValue
 import com.amazon.ion.system.IonSystemBuilder
 import com.amazon.ionschema.IonSchemaSystemBuilder
 import com.amazon.ionschema.Schema
+import org.partiql.lang.eval.ExprValue
 import org.partiql.lang.partiqlisl.getResourceAuthority
 import java.io.File
 
 class GraphValidationException(message: String) : RuntimeException(message)
+class GraphReadException(message: String) : RuntimeException(message)
 
-/** A validator for external graphs represented in Ion in accordance with the graph.isl schema.
+internal typealias EdgeTriple<E> = Triple<SimpleGraph.Node, E, SimpleGraph.Node>
```

Comment:
Nit, internal so doesn't really matter but an upper bound on `E` would be nice.

I can't tell if `E` is for Edge or Element. Either way
```suggestion
internal typealias EdgeTriple<E : Edge> = Triple<SimpleGraph.Node, E, SimpleGraph.Node>
```

Reply:
We think similarly! -- I tried this before.  Alas, "Bounds are not allowed on type alias parameters".